.github/workflows/ci.yml

```diff
@@ -13,7 +13,9 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
 permissions: read-all
 env:
-  TRITON_BUILD_WITH_CCACHE: "true"
+  # Disable ccache as it seemed to have weird behavior and caused significant slowdown.
+  # Reenable if needed
+  # TRITON_BUILD_WITH_CCACHE: "true"
   TRITON_BUILD_WITH_CLANG_LLD: "TRUE"
```

Comment:
@peterbell10 I think I was wrong and @ThomasRaoux was completely correct. The real win here was caused by disabling ccache actually, so I vote for disabling it for now. ~25 mins isn't too bad considering that sometimes AMD CI can run up to 25 mins as well.

Reply:
How did you determine that ccache is causing a regression in perf?

Reply:
It's based on experimental observation. 

In the current CI, we always use ccache (without lld), and it sometimes causes macos-build to last for 60-70 minutes.

After I take it out and try for several runs, it seems like macos-build can complete in 25 minutes.

Reply:
hmm, interesting.

There are cases though where we do hit cache but still see significant time.

Reply:
Another possible explanation is that the macos node from github is unstable

Reply:
We can see from the logs that the build is getting stalled during linking and `ccache` doesn't effect linking so it seems unrelated. Regarding the log you provided, it seems to stall for half an hour on linking tests?
```
 [356/389, 546.150s elapsed] Linking CXX executable third_party/nvidia/unittest/Conversion/TritonGPUToLLVM/TestPtxAsmFormat
  [357/389, 547.752s elapsed] Linking CXX executable third_party/amd/unittest/Conversion/TestOptimizeLDS
  [358/389, 2687.154s elapsed] Building CXX object bin/CMakeFiles/triton-opt.dir/triton-opt.cpp.o
```

Reply:
If you really think it's due to ccache then I suppose it doesn't hurt to try without it for a while and see how it goes, but it would be very surprising to me if it's the cause of the slowdown.